### ISSUE TITLE
tclfmt doesn't catch illegal expression bareword syntax error gracefully

### ISSUE THREAD

**Issue by nmoroze** [nmoroze/tclint#148]
```
$ echo "expr asdf" | tclfmt -                                                                                                          
Traceback (most recent call last):
  File "/Users/noah/Code/tclint/src/tclint/cli/tclfmt.py", line 148, in main
    formatted = format(
        script,
    ...<3 lines>...
        partial=args.partial,
    )
  File "/Users/noah/Code/tclint/src/tclint/cli/tclfmt.py", line 44, in format
    return formatter.format_top(script, parser)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/Users/noah/Code/tclint/src/tclint/format.py", line 152, in format_top
    tree = parser.parse(script)
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 127, in parse
    tree = self._parse_script(lexer, in_command_sub=False)
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 216, in _parse_script
    cmd = self.parse_command(ts, in_command_sub=in_command_sub)
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 287, in parse_command
    parsed_args = self._parse_command_args(routine.contents, args)
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 165, in _parse_command_args
    raise e
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 163, in _parse_command_args
    new_args = check_command(routine, args, self, spec)
  File "/Users/noah/Code/tclint/src/tclint/commands/checks.py", line 145, in check_command
    return command_spec(args, parser)
  File "/Users/noah/Code/tclint/src/tclint/commands/builtin.py", line 381, in _expr
    return [parser.parse_expression(args[0])]
            ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 601, in parse_expression
    contents = self._parse_expression(ts)
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 63, in func
    node = parse_func(parser, ts)
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 614, in _parse_expression
    op1 = self._parse_operand(ts)
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 63, in func
    node = parse_func(parser, ts)
  File "/Users/noah/Code/tclint/src/tclint/parser.py", line 725, in _parse_operand
    raise TclSyntaxError(
        f"invalid bareword in expression: {operand}", operand_pos, ts.pos()
    )
tclint.lexer.TclSyntaxError: invalid bareword in expression: asdf

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/noah/Code/tclint/.venv/bin/tclfmt", line 10, in <module>
    sys.exit(main())
             ~~~~^^
  File "/Users/noah/Code/tclint/src/tclint/cli/tclfmt.py", line 171, in main
    line, col = e.pos
                ^^^^^
AttributeError: 'TclSyntaxError' object has no attribute 'pos'
```
